tests/substrate_tests/builtins.rs

```diff
@@ -725,6 +725,11 @@ fn hash() {
             function get() public view returns (Hash) {
                 return Hash.wrap(current2);
             }
+
+            function test_encoding() public view {
+                Hash h = Hash.wrap(current2);
+                assert(abi.encode(current2) == abi.encode(h));
```

Comment:
Be careful with tests like this. They might mask a case when both the encoding and decoding are wrong.